wallets/provider-phantom/src/namespaces/evm.ts

```diff
@@ -0,0 +1,37 @@
+import type { EvmActions } from '@rango-dev/wallets-core/namespaces/evm';
+
+import { NamespaceBuilder } from '@rango-dev/wallets-core';
+import { utils } from '@rango-dev/wallets-core/namespaces/common';
+import {
+  actions,
+  after,
+  and,
+  before,
+} from '@rango-dev/wallets-core/namespaces/evm';
+
+import { WALLET_ID } from '../constants.js';
+import { evmPhantom } from '../legacy/helpers.js';
+
+const [changeAccountSubscriber, changeAccountCleanup] =
+  actions.changeAccountSubscriber(evmPhantom);
+
+const evm = new NamespaceBuilder<EvmActions>('evm', WALLET_ID)
+  .action('init', () => {
+    console.log('[phantom]init called from evm cb');
+  })
+  .action([...actions.recommended, actions.connect(evmPhantom)])
+  .andUse(and.recommended)
+  .build();
```

Comment:
As we've talked offline, each action can be imported from `core` and pass one by one using `.action` to builder. This is the explicit way and we already have support for it, another approach is use `recommended` values from `core` which basically are actions that can be used in all providers. they are crafted and maintained carefully so without needing any extra efforts to update provider package (e.g phantom), it can receives some recommended actions at anytime easily.

Reply:
This is my suggestion on how you could improve this approach by having `before`, `after` and other hooks composed with action in single object instead of separating them and importing duplicate stuff everywhere.

const connect = new ActionBuilder('connect', () => {})
   .('after', () => {})
   .('before', () => {})
   .('and', () => {})

const evm = new NamespaceBuilder<EvmActions>('evm', WALLET_ID)
  .action('init', () => {})
  .action([connect])

Reply:
I created a new builder called `ActionBuilder` to achieve what you've proposed here. I only applied it to `evm` namespace for Phantom. If the approach has been finalized, I will update solana namespace for phatom as well.